probe-rs-tools/src/bin/probe-rs/cmd/dap_server/peripherals/svd_cache.rs

```diff
@@ -247,19 +251,32 @@ impl SvdVariable {
                         address
                     )
                 } else {
-                    let value = match memory.read_word_32(*address) {
-                        Ok(u32_value) => Ok(u32_value),
-                        Err(error) => Err(format!(
-                            "Unable to read peripheral register value @ {:#010X} : {:?}",
-                            address, error
-                        )),
+                    let size_bytes = (*size / 8).max(1);
+                    let bits_alignment_offset = (*address % size_bytes as u64) as u32 * 8;
+                    let value = match *size {
+                        0..=8 => memory
+                            .read_word_8(*address)
+                            .map(|val| format!("{val:#04X}")),
+                        9..=16 => {
+                            let aligned_address = *address & !0b1;
+                            memory
+                                .read_word_16(aligned_address)
+                                .map(|val| format!("{:#06X}", val >> bits_alignment_offset))
+                        }
+                        _ => {
```

Comment:
I think we should also handle the 64-bit case, I'm not sure if we have targets but someone wanted 64-bit RTT code so someone may have one :)

Reply:
The current code only support up to 32bits anyway, so I think we could open an issue to track proper support of 64bits devices.
Since I don't have any, and this PR is here to fix bad behavior of currently supported devices, I think we should not bother with that here.